### PR TITLE
Ohm 443 create load tests

### PR DESCRIPTION
**Pull request by napiergit**

### CODE REVIEW

File: test/load-tests/load-test-utils.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      const tokenhash = crypto.createHash('sha512')
+      tokenhash.update(passwordHash)
+      tokenhash.update(requestSalt)
+      tokenhash.update(authTS)
+      
+      return {
+        'auth-username': username,
+        'auth-ts': authTS,
+        'auth-salt': requestSalt,
+        'auth-token': tokenhash.digest('hex')
```
Comment (napiergit) [inline test/load-tests/load-test-utils.js]: This is not authenticating for some reason.

File: test/load-tests/env/start-server.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+docker cp ./test/load-tests/env/dump.tar.gz openhim-mongo:/ &&
+docker exec -it openhim-mongo bash -c "tar -xzvf dump.tar.gz dump ; mongorestore /dump" &&
```
Comment (BMartinos) [inline test/load-tests/env/start-server.sh]: We dont seem to do anything with the mongo container, so the data will not exist when running the load tests

File: test/load-tests/api.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  phases:
+    - duration: 10
+      name: 'Low load'
+      arrivalRate: 3
+    # - duration: 10
+    #   name: 'Medium load'
+    #   arrivalRate: 20
+    # - duration: 10
+    #   name: 'High load'
+    #   arrivalRate: 50
```
Comment (BMartinos) [inline test/load-tests/api.yml]: Lets un-comment this so that we can test the load a bit

**Comment by BMartinos** [comment]
Sorry last thing i missed in the comments, When running the clients load tests, the responses are all 500, its trying to reach an port that doesnt have anything running on it, if we can somehow get it to respond with 200 status code, maybe creating a small service listening on that port

File: package.json [hunk 4]
```diff
@@ -20,7 +20,10 @@
     "test": "cross-env NODE_ENV=test NODE_TLS_REJECT_UNAUTHORIZED=0 mocha -t 16000 --require babel-register test/**/*.js",
     "test:unit": "cross-env NODE_ENV=test mocha --require babel-register test/unit/**/*.js",
     "test:int": "cross-env NODE_TLS_REJECT_UNAUTHORIZED=0 mocha -t 4000 --require babel-register test/integration/**/*.js",
-    "start": "node --harmony lib/server.js"
+    "test:load:client": "./test/load-tests/env/start-server.sh && sleep 5 && artillery run -o artillery/load-clients.json test/load-tests/clients.yml && npm stop && artillery report artillery/load-clients.json",
```
Comment (napiergit) [inline package.json]: added as dev dependency

File: test/load-tests/api.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  processor: './load-test-utils.js'
+      
+scenarios:
+  - flow:
+    - get:
+        url: "/users"
+        beforeRequest: 'setAuthHeaders'
```
Comment (napiergit) [inline test/load-tests/api.yml]: image from the docs.

**Pull request merged**